### DISCUSSION TITLE
Composite Resource generation using Composition Fucntions

### DISCUSSION THREAD

**Discussion by smithapatankar61286601** [crossplane/crossplane#4548] [Q&A] [answered]
Problem Statement:
I need to generate Composition Resources in Composition based on the inputs provided at runtime.

Example:
If the input file contains inputs for 3 subnets, then Composition must contain 3 Subnet Composite Resources, one for each subnet.
Can this be acheived using Composition functions?
If yes, could you please share some examples?

**Comment by negz** [comment, marked as answer]
If I'm following you correctly, this is possible. This lab may help - https://github.com/crossplane-contrib/contribfest/tree/main/lab-composition-functions

**Reply by smithapatankar61286601** [reply]
This is exactly what I was looking for the use case I have - https://github.com/crossplane-contrib/contribfest/blob/main/lab-composition-functions/04-xfn-many.md
Thank you, @negz!

**Reply by smithapatankar61286601** [reply]
Hi @negz , 

Is it already known in which release of Crossplane will "Composition Function" feature be promoted to v1/GA?

**Reply by jbw976** [reply]
Not yet @spatanka10 - CompositionFunctions are expected to move to Beta in v1.14 (late October) - you can follow along with the progress in https://github.com/crossplane/crossplane/issues/3751.

A date for GA is not scheduled yet.  Beta functionality tends to be stable and typically ready for production usage. You can read more about that in https://docs.crossplane.io/knowledge-base/guides/feature-lifecycle/#beta-features.

**Reply by smithapatankar61286601** [reply]
Thank you @jbw976 .
